alphastats/gui/pages/02_Import Data.py

```diff
@@ -1,76 +1,141 @@
+st.write(
+    "Start a new session will discard the current one (including all analysis!) and enable importing a new dataset."
+)
+st.write("To explore AlphaPeptStats you may also load an example dataset.")
 
-if "loader" not in st.session_state:
-    st.session_state["loader"] = None
+c1, c2 = st.columns(2)
+if c1.button("Start new Session"):
+    empty_session_state()
+    st.rerun()
 
-if "gene_to_prot_id" not in st.session_state:
-    st.session_state["gene_to_prot_id"] = {}
+if c2.button("Start new Session with example DataSet"):
+    empty_session_state()
+    init_session_state()
+    loader, metadata_columns, dataset = load_example_data()
 
-if "organism" not in st.session_state:
-    st.session_state["organism"] = 9606  # human
+    st.session_state["dataset"] = dataset
+    st.session_state["metadata_columns"] = metadata_columns
+    st.session_state["loader"] = loader
+    load_options()
+    # TODO why are we doing this so early?
+    save_plot_sampledistribution_rawdata(dataset)
+    sidebar_info()
+    st.stop()
 
-sidebar_info()
 
-if "dataset" not in st.session_state:
-    st.markdown("### Import Proteomics Data")
+st.markdown("### Import Proteomics Data")
+if "dataset" in st.session_state:
+    st.info(f"DataSet already present: {st.session_state['dataset']}")
+    st.stop()
 
-    st.markdown(
-        "Create a DataSet with the output of your proteomics software package and the corresponding metadata (optional). "
-    )
 
-import_data()
+st.markdown(
+    "Create a DataSet with the output of your proteomics software package and the corresponding metadata (optional). "
+)
 
-if "dataset" in st.session_state:
-    st.info("DataSet has been imported")
+# ########## Select Software
+st.markdown("##### 1. Select software and upload data")
 
-    if "distribution_plot" not in st.session_state:
-        save_plot_sampledistribution_rawdata()
+default_select_option = "<select>"
+options = [default_select_option] + list(SOFTWARE_OPTIONS.keys())
 
-    display_loaded_dataset()
+software = st.selectbox(
+    "Select your Proteomics Software",
+    options=options,
+)
+if software == default_select_option:
+    st.stop()
 
-st.markdown("### Or Load sample Dataset")
 
-if st.button("Load sample DataSet - PXD011839", key="load_sample_data"):
-    load_sample_data()
-    if "distribution_plot" not in st.session_state:
-        save_plot_sampledistribution_rawdata()
+# ########## Load Software File
 
+softwarefile = st.file_uploader(
+    SOFTWARE_OPTIONS.get(software).get("import_file"),
+    type=["csv", "tsv", "txt", "hdf"],
+)
 
-st.markdown("### To start a new session:")
+if softwarefile is None:
+    st.stop()
 
-if st.button("New Session: Import new dataset"):
-    empty_session_state()
-    st.rerun()
+softwarefile_df = load_softwarefile_df(software, softwarefile)
+
+intensity_column, index_column = show_loader_columns_selection(
+    software=software, softwarefile_df=softwarefile_df
+)
+
+loader = load_proteomics_data(
```

Comment:
It could be that we actually need some logic here for the custom upload, which serves the multiselect instead of selector. Intensity columns would then be None initially.